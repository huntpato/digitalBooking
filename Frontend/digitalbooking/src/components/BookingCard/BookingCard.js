import React, { useContext } from "react";
import { Link } from 'react-router-dom'
import { UserContext } from "../UserContext/UserContext";
import styles from "./BookingCard.module.css";

export const BookingCard = (props) => {
    const { user } = useContext(UserContext);
    const { path }= props;
    const { bookingCards, resButton } = styles;
    let component = user ? <Link to={`../reservation/${path}`} className={resButton}><button>Iniciar reserva</button></Link> : <Link to={`../login/resFailed`} className={resButton}><button>Iniciar reserva</button></Link> ;
    return (
      <article className={ bookingCards }>
          <p>Agreg√° tus fechas de viaje para obtener precios exactos</p>
         {component}
      </article>
    );
  };