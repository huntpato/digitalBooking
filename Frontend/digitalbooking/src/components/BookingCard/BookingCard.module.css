.bookingCards{

    align-self: center;
    background-color: white;
    border-radius: 5px;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
    display:flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-left:40px;
    width:23%;
    height:110px;

}

.bookingCards button{
    background-color: transparent;
    border: none;
    color: var(--background-white);
    cursor: pointer;
}

.resButton{

    align-items: center;
    background-color: var(--principal-color);
    border-radius: 5px;
    cursor: pointer;
    display: flex;
    font-weight: 700;
    justify-content: center;
    height: 30px;
    text-decoration: none;
    width: 90%;
}

.bookingCards p{
    font-weight: 600;
    margin: 0px 0px 15px 0px;
    width: 90%;
    text-align: center;
}

@media (max-width: 1046px){

    .bookingCards{

        margin-top: 30px;
        margin-left:0px;
        width:50%;
    
    }

    .bookingCards button{

        width: 75%;
    }
    
    .bookingCards p{
        font-weight: 600;
        margin: 0px 0px 15px 0px;
        width: 90%;
        text-align: center;
    }
    

}

@media (max-width: 768px){

    .bookingCards{

        align-self: center;
        background-color: transparent;
        border-radius: 0px;
        box-shadow: 0px 0px 0px rgba(0, 0, 0, 0.15);
        display:flex;
        flex-direction: row;
        align-items:flex-end;
        justify-items: space-between;
        margin-left:0px;
        width:100%;
        height:80px;
        
    
    }

    .bookingCards button{

        width: 50%;
    }

    .bookingCards p{

        margin: 0px 10px 0px 0px;
        width: 50%;
        text-align: left;
    }

}


@media (max-width: 480px){

    .bookingCards{
        align-self: center;
        flex-direction: column;
        align-items:center;
        width:100%;
        height:100px;
    }

    .bookingCards button{
        width: 90%;
    }

    .bookingCards p{
        margin: 0px 0px 10px 0px;
        width: 90%;
        text-align: center;
    }

}