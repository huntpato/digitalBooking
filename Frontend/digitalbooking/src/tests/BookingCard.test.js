import React from 'react';
import { shallow } from 'enzyme';
import { BookingCard } from '../components/BookingCard/BookingCard'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <BookingCard />', ()=> {

    
    const wrapper= shallow(<BookingCard />);

    test('Comprobar que BookingCard renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    
})
