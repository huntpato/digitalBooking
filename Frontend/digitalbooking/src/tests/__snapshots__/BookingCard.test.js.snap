// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <BookingCard /> Comprobar que BookingCard renderiza correctamente. 1`] = `
<article
  className="bookingCards"
>
  <p>
    Agreg√° tus fechas de viaje para obtener precios exactos
  </p>
  <Link
    className="resButton"
    to="../login/resFailed"
  >
    <button>
      Iniciar reserva
    </button>
  </Link>
</article>
`;
